Http/Http.cs

```diff
@@ -0,0 +1,352 @@
+		/// <param name="uri">The URI to which a DELETE request should be sent.</param>
+		/// <returns>
+		///   <para>A HttpRequest configured to send an HTTP DELETE request.</para>
+		/// </returns>
+		public static HttpRequest Delete(string uri)
+		{
+			return new HttpRequest(UnityWebRequest.Delete(uri));
+		}
+
+		/// <summary>
+		///   <para>Creates a HttpRequest configured to send a HTTP HEAD request.</para>
+		/// </summary>
+		/// <param name="uri">The URI to which to send a HTTP HEAD request.</param>
+		/// <returns>
+		///   <para>A HttpRequest configured to transmit a HTTP HEAD request.</para>
+		/// </returns>
+		public static HttpRequest Head(string uri)
+		{
+			return new HttpRequest(UnityWebRequest.Head(uri));
+		}
+
+		#endregion
+
+		#region None Static Send Methods
+
+		public void Send(HttpRequest request, Action<HttpResponse> onSuccess = null, Action<HttpResponse> onError = null)
+		{
+			StartCoroutine(SendCoroutine(request, onSuccess, onError));
+		}
+
+		public void Send(UnityWebRequest unityWebRequest, Action<UnityWebRequest> onSuccess = null,
+			Action<UnityWebRequest> onError = null, Action<UnityWebRequest> onNetworkError = null)
+		{
+			StartCoroutine(SendCoroutine(unityWebRequest, onSuccess, onError, onNetworkError));
+		}
+
+		public void Send(UnityWebRequest unityWebRequest, Action onSuccess = null,
+			Action onError = null, Action onNetworkError = null)
+		{
+			StartCoroutine(SendCoroutine(unityWebRequest, onSuccess, onError, onNetworkError));
+		}
+
+		public void Send(UnityWebRequest unityWebRequest, Action<HttpResponse> onSuccess = null,
+			Action<HttpResponse> onError = null)
+		{
+			StartCoroutine(SendCoroutine(unityWebRequest, onSuccess, onError));
+		}
+
+		#endregion
+
+		#region Send HttpRequest methods
+
+		private static IEnumerator SendCoroutine(HttpRequest request, Action<HttpResponse> onSuccess,
+			Action<HttpResponse> onError)
+		{
+			var unityWebRequest = request.UnityWebRequest;
+			yield return unityWebRequest.SendWebRequest();
+			var response = new HttpResponse(unityWebRequest);
+
+			if (onError != null && (unityWebRequest.isNetworkError || unityWebRequest.isHttpError))
+			{
+				onError.Invoke(response);
+			}
+			else if (onSuccess != null)
+			{
+				onSuccess.Invoke(response);
+			}
+		}
+
+		#endregion
+
+		#region Send UnityWebRequest methods
+
+		private static IEnumerator SendCoroutine(UnityWebRequest unityWebRequest,
+			Action onSuccess = null, Action onError = null, Action onNetworkError = null)
+		{
+			yield return unityWebRequest.SendWebRequest();
+			if (unityWebRequest.isNetworkError)
+			{
+				if (onNetworkError != null)
+				{
+					onNetworkError.Invoke();
+				}
+			}
+			else if (unityWebRequest.isHttpError)
+			{
+				if (onError != null)
+				{
+					onError.Invoke();
+				}
+			}
+			else
+			{
+				if (onSuccess != null)
+				{
+					onSuccess.Invoke();
+				}
+			}
+		}
+
+		private static IEnumerator SendCoroutine(UnityWebRequest unityWebRequest,
```

Comment:
I may be blind, but I am not seeing any difference between these methods? Their signatures are the same?

Reply:
One of the methods does have an additional
_Action<UnityWebRequest> onError = null, Action<UnityWebRequest> onNetworkError = null)
at the end. It's inline with another parameter, that's probably why you didn't see it.

Reply:
Depends which one you are referring too, Kevin is correct if referring to the method below, however if referring to the method above then the differences is the signature of the callback methods, the first one is
Action<UnityWebRequest> and the second is Action. Its to cater to both situations where you either have the UnityWebRequest cached already or you don't, so it gets given back to you.

Reply:
Weird, the Action type got formatted out in your comment Harry, it should say one is `Action<UnityWebRequest>`